### ISSUE TITLE
wrong accuracy

### ISSUE THREAD

**Issue by lbj0411** [dmlc/mxnet#955]
I try to train my own data with the mnist net.The data has 9 classes(with label 0,1,2...8)and 7500+ .jpg images(28*28 ) for training and 760 .jpg images for validation.

When I use the caffe, I could get 90% val-accuracy after 10000 iterations(batch_size = 64),but when I use the mxnet with the same network, the validation accuracy is around 0.12(maybe related to 1/9) since the beginning.My learning rate is 0.0001,and I definitely shuffle the data txt(which I wrote with matlab, not use the make_list.py).And I have changed the initializer = mx.int.Normal(sigma = 0.01),the result is low the same as mentioned above.

Another problem is my training images is all gray images. But when I define the data_shape(1,28,28),it will return mismatch error.Only with (3,28,28),it could run.

Do you guys have any idea where am I doing wrong?

Here is my training script:

    import find_mxnet
    import mxnet as mx
    import argparse
    import os, sys
    import train_model
    
    parser = argparse.ArgumentParser(description='train an image classifer on mnist')
    parser.add_argument('--network', type=str, default='lenet',
                        choices = ['mlp', 'lenet'],
                        help = 'the cnn to use')
    parser.add_argument('--data-dir', type=str, default='emotion/',
                        help='the input data directory')
    parser.add_argument('--gpus', type=str, default='0',
                        help='the gpus will be used, e.g "0,1,2,3"')
    parser.add_argument('--num-examples', type=int, default=7534,
                        help='the number of training examples')
    parser.add_argument('--batch-size', type=int, default=64,
                        help='the batch size')
    parser.add_argument('--lr', type=float, default=.0001,
                        help='the initial learning rate')
    parser.add_argument('--model-prefix', type=str,default='emotion/result/',
                        help='the prefix of the model to load/save')
    parser.add_argument('--num-epochs', type=int, default=200,
                        help='the number of training epochs')
    parser.add_argument('--load-epoch', type=int,
                        help="load the model on an epoch using the model-prefix")
    parser.add_argument('--kv-store', type=str, default='local',
                        help='the kvstore type')
    parser.add_argument('--lr-factor', type=float, default=1,
                        help='times the lr with a factor for every lr-factor-epoch epoch')
    parser.add_argument('--lr-factor-epoch', type=float, default=1,
                        help='the number of epoch to factor the lr, could be .5')
    args = parser.parse_args()

    def get_lenet():
        """
        LeCun, Yann, Leon Bottou, Yoshua Bengio, and Patrick
        Haffner. "Gradient-based learning applied to document recognition."
        Proceedings of the IEEE (1998)
        """
        data = mx.symbol.Variable('data')
        # first conv
        conv1 = mx.symbol.Convolution(data=data, kernel=(5,5), num_filter=20)
        pool1 = mx.symbol.Pooling(data=conv1, pool_type="max",
                                  kernel=(2,2), stride=(2,2))
        # second conv
        conv2 = mx.symbol.Convolution(data=pool1, kernel=(5,5), num_filter=50)
        pool2 = mx.symbol.Pooling(data=conv2, pool_type="max",
                                  kernel=(2,2), stride=(2,2))
        # first fullc
        flatten = mx.symbol.Flatten(data=pool2)
        fc1 = mx.symbol.FullyConnected(data=flatten, num_hidden=500)
        relu1 = mx.symbol.Activation(data=fc1, act_type="relu")
        # second fullc
        fc2 = mx.symbol.FullyConnected(data=relu1, num_hidden=9)
        # loss
        lenet = mx.symbol.SoftmaxOutput(data=fc2, name='softmax')
        return lenet
    
    net = get_lenet()

    def get_iterator(args, kv):
        data_dir = args.data_dir
        data_shape = (3, 28, 28)
        train = mx.io.ImageRecordIter(
            path_imgrec = data_dir + "train.rec",
            data_shape  = data_shape,
            batch_size  = args.batch_size,
            shuffle     = True,
            rand_crop   = True,
            rand_mirror = True,
            num_parts   = kv.num_workers,
            part_index  = kv.rank)

        val = mx.io.ImageRecordIter(
            path_imgrec = data_dir + "val.rec",
            data_shape  = data_shape,
            batch_size  = args.batch_size,
            shuffle     = True,
            rand_crop   = False,
            rand_mirror = False,
            num_parts   = kv.num_workers,
            part_index  = kv.rank)
    
        return (train, val)
    
    # train
    train_model.fit(args, net, get_iterator)

**Comment by tqchen** [comment]
You can try debugging by start with the simplest model, for example, 1 layer softmax to see if things converges, and add layers gradually

**Comment by lbj0411** [comment]
I find where is wrong,thank you!
